src/main/java/org/takes/facets/auth/PsBasic.java

```diff
@@ -185,4 +195,39 @@ public Fake(final boolean cond) {
             return new Opt.Empty<Identity>();
         }
     }
+
+    /**
+     * Default implementation. Using list of user and pass.
+     * @author lcozzani (lautaromail@gmail.com)
+     * @version $Id$
+     * @since 0.22
+     */
+    public static final class Default implements PsBasic.Entry {
+
+        /**
+         * Map of users and its passwords.
+         */
+        private final transient Map<String, String> users;
+
+        /**
+         * Ctor.
+         * @param usrs Map of user and pass
+         */
+        public Default(final Map<String, String> usrs) {
+            super();
+            this.users = usrs;
+        }
+
+        @Override
+        public Opt<Identity> enter(final String user, final String pwd) {
+            Opt<Identity> valid = new Opt.Empty<Identity>();
```

Comment:
@lautarobock let's assign `new Opt.Empty` statement into an `else` block instead of initializing the variable here. That way we can a) declare `valid` as `final`, and b) avoid instantiating `Opt.Empty` if unnecessary.